lib/babe/build_integration_test.go

```diff
@@ -139,6 +139,49 @@ func TestBuildBlock_ok(t *testing.T) {
 	require.Equal(t, 1, len(extsBytes))
 }
 
+func TestBuildBlock_addToTransactionState(t *testing.T) {
```

Comment:
Can you add to this test a push before the slot timer expires?

Right now the block

```go
			case <-pushWatcher:
				txn = b.transactionState.Pop()
				// check in case another goroutine popped the
				// transaction before the Pop() call above.
				retry = txn == nil
```

is not covered.

Ideally I'd say the following pattern would be nice to integrate test this:

1. Get item from the push watcher
1. Get item from queue directly
1. Get item from the push watcher
1. Slot timer expires so we exit (either at top of loop or within the push watcher select block)

Reply:
Created test `TestBuildBlock_buildBlockExtrinsics` which now covers above block.